src/main/java/mate/academy/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add user " + user, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from User u where u.email = :email", User.class)
```

Comment:
```suggestion
            return session.createQuery("from User u where u.email = :email", User.class)
```
Alias is redundant here (it’s more appropriate to query with join) but you can leave  it if you want